iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/stack.inl

```diff
@@ -22,6 +22,122 @@ namespace iox
 {
 namespace cxx
 {
+template <typename T, uint64_t Capacity>
+inline stack<T, Capacity>& stack<T, Capacity>::copy(const stack& rhs) noexcept
+{
+    uint64_t i{0};
+    const uint64_t rhsSize{rhs.size()};
+    const uint64_t minSize{algorithm::minVal(m_size, rhsSize)};
+
+    // copy assignment
+    for (; i < minSize; i++)
+    {
+        getUnchecked(i) = rhs.getUnchecked(i);
+    }
+    // copy c'tor
+    for (; i < rhsSize; i++)
+    {
+        // AXIVION Next Line AutosarC++19_03-A18.5.2 : false positive, it is a placement new
+        new (&m_data[i]) T(rhs.getUnchecked(i));
+    }
+    // delete remaining elements
+    clearFrom(i);
+
+    m_size = rhsSize;
+
+    return *this;
+}
+
+template <typename T, uint64_t Capacity>
+inline stack<T, Capacity>& stack<T, Capacity>::move(stack&& rhs) noexcept
+{
+    uint64_t i{0};
+    const uint64_t rhsSize{rhs.size()};
+    const uint64_t minSize{algorithm::minVal(m_size, rhsSize)};
+
+    // move assignment
+    for (; i < minSize; i++)
+    {
+        getUnchecked(i) = std::move(rhs.getUnchecked(i));
+    }
+    // move c'tor
+    for (; i < rhsSize; i++)
+    {
+        // AXIVION Next Line AutosarC++19_03-A18.5.2 : false positive, it is a placement new
+        new (&m_data[i]) T(std::move(rhs.getUnchecked(i)));
+    }
+    // delete remaining elements
+    clearFrom(i);
+
+    m_size = rhsSize;
+    rhs.clear();
+
+    return *this;
+}
+
+// AXIVION Next Construct AutosarC++19_03-A12.6.1 : the remaining m_data fields are explicitly initialized when a new
+// element is pushed
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init, hicpp-member-init)
+template <typename T, uint64_t Capacity>
+inline stack<T, Capacity>::stack(const stack& rhs) noexcept
+{
+    copy(rhs);
```

Comment:
Why did you introduce `copy` and `move` methods and do not use `*this = rhs` or `*this = std::move(rhs)`

Reply:
To avoid the self assignment check in the c'tors.